packages/request-client.js/src/api/payment-network/erc20/proxy-contract.ts

```diff
@@ -0,0 +1,74 @@
+import { AdvancedLogicTypes, ExtensionTypes, RequestLogicTypes } from '@requestnetwork/types';
+import Utils from '@requestnetwork/utils';
+import * as Types from '../../../types';
+
+/**
+ * Handle payment networks with ERC20 proxy contract extension
+ */
+export default class PaymentNetworkERC20ProxyContract implements Types.IPaymentNetwork {
+  private extension: ExtensionTypes.PnReferenceBased.IReferenceBased;
+  /**
+   * @param extension The advanced logic payment network extensions
+   */
+  public constructor({ advancedLogic }: { advancedLogic: AdvancedLogicTypes.IAdvancedLogic }) {
+    this.extension = advancedLogic.extensions.erc20ProxyContract;
```

Comment:
not familiar with how this works, but maybe we should test that it's defined here?

Reply:
extension is a private member, that's why we doesn't test it but your comment made me creating this task: https://requestnetwork.atlassian.net/browse/PROT-1047